ratpack-session/src/main/java/ratpack/session/internal/JavaBuiltinSessionSerializer.java

```diff
@@ -40,4 +41,36 @@
       }
     }
   }
+
+  private static final class TypeFilteringObjectOutputStream extends ObjectOutputStream {
+    private final SessionTypeFilter typeFilter;
+
+    public TypeFilteringObjectOutputStream(OutputStream out, SessionTypeFilter typeFilter) throws IOException {
+      super(out);
+      this.typeFilter = typeFilter;
+    }
+
+    @Override
+    protected void writeClassDescriptor(ObjectStreamClass desc) throws IOException {
+      typeFilter.assertAllowed(desc.forClass());
+      super.writeClassDescriptor(desc);
+    }
+  }
+
+  private static final class TypeFilteringObjectInputStream extends ObjectInputStream {
```

Comment:
Looks like this is a less feature-rich version of the `ValidatingObjectInputStream` from Apache Commons.

I just want to make sure that you're not reinventing the wheel here.

https://github.com/apache/commons-io/blob/master/src/main/java/org/apache/commons/io/serialization/ValidatingObjectInputStream.java

Reply:
We don't need any of those features, or the extra dependency.